resources/web/docs_js/index.js

```diff
@@ -267,5 +267,23 @@ $(function() {
     document.head.appendChild(hotcss);
   }
 
+  // If you're have been given access to the private repository these docs
+  // relate to *and* you use the 'edit' in the query string or hash, we'll show
+  // you an edit button taking you to the private repo.
+  if (window.location.search.indexOf('edit') > -1 || window.location.hash.indexOf('edit') > -1) {
```

Comment:
I think we should explicitly test for `?edit` in the hash, because I suspect we often use hashes to link to sections of the documentation, and there is a high chance the word `edit` would be there already. The `?edit` on the other hand is much less likely.  I think having `edit` as is in the query part is ok, as I don't think we use query params (unless we use it for searching in the docs?)

Reply:
> there is a high chance the word `edit` would be there already

agreed - amended

>  I don't think we use query params (unless we use it for searching in the docs?)

I asked myself that a lot as I wondered whether the param preserving code was needed! Best to assume we do. Given that with this PR we have the ability to `var query_obj = utils.deparam(this.search);` maybe just use that and check for the presence of `query_obj.edit`?

Reply:
Now using URLSearchParams web API to achieve this with polyfill for IE, it works very well.